enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -182,4 +191,70 @@ impl<'a> Handler<'a> {
             ret
         }
     }
+
+    /// Do a readv() syscall
+    pub fn readv(&mut self) -> u64 {
+        let fd = self.aex.gpr.rdi;
+        let trusted = unsafe {
+            from_raw_parts_mut(self.aex.gpr.rsi as *mut Iovec, self.aex.gpr.rdx as usize)
+        };
+
+        // Add up total size of buffers.
+        let bufsize = trusted.iter().fold(0, |a, e| a + e.size);
+
+        // Allocate some unencrypted memory.
+        let size = bufsize + size_of::<Iovec>() * trusted.len();
+        let map = match self.ualloc(size as u64) {
+            Err(errno) => return errno.into_syscall(),
+            Ok(map) => unsafe { from_raw_parts_mut(map, size as usize) },
+        };
+
+        // Split allocated memory into that used by the iovec struct array and that used by its buffers.
+        let (uiovec, ubuffer) = map.split_at_mut(size_of::<Iovec>() * trusted.len());
+
+        // Convert the prefix from a byte slice into an iovec slice.
+        let (pre, untrusted, suf) = unsafe { uiovec.align_to_mut::<Iovec>() };
+        if pre != [] || suf != [] {
+            self.attacked();
+        }
```

Comment:
```suggestion
        let (_, untrusted, _) = unsafe { uiovec.align_to_mut::<Iovec>() };
        if untrusted.len() != trusted.len() {
            self.attacked();
        }
```

Reply:
This does make way more sense as a method of checking the same thing. Thank you.